build(docs): add a script for fast previews of LB4 doc changes

bajtos opened the pull request:
Add a script `docs/bin/build-preview-site.sh` that creates a subset of loopback.io site with just enough content to support fast previews of LB4 documentation. Add monorepo-level npm scripts to simplify usage of the new build script.

Fixes #1310

Usage:

1. Run the script to create a new Jekyll site in `docs/_preview` dir:

    $ npm run docs:prepare

2. Follow the instructions printed by the script to start Jekyll:

    $ npm run docs:start

3. View the documentation.

4. Make changes to documentation files in `docs/site`.

5. Restart Jekyll (stop the running process and go to step 2).

The good parts:

- Initial build of LB4-only docs takes about 6 seconds. 🏇 (Compare to ~4 minutes for the entire docs site! 😱)

- Incremental rebuild after changing a single file takes less than a second. 🏎

Downsides:

- Watch mode does not work. AFAICT, Jekyll cannot watch symlinked files stored outside of `source` directory. We can investigate this later.


<!--
Please provide a high-level description of the changes made by your pull request.

Include references to all related GitHub issues and other pull requests, for example:

Fixes #123
Implements #254
See also #23
-->

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- Affected artifact templates in `packages/cli` were updated
- Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

bajtos commented:
Discussion point:

Do we want to add scripts to the top-level (monorepo-level) package.json, one for setting up the preview and another for running Jekyll?

Once we agree on the high-level direction, I'll update developer documentation to describe how to use this new preview tooling.

raymondfeng commented on docs/bin/build-jekyll-preview-config.js:
```diff
@@ -0,0 +1,29 @@
+  console.log('Usage:');
+  console.log();
+  console.log('  node %s <source _config.yml> [dest path]', process.argv[1]);
+  console.log();
```
Is it simpler to use the following?
```js
console.log(`Missing required argument: path to original Jekyll config

Usage:

  node ${process.argv[1]} <source _config.yml> [dest path]
`);
```

bajtos replied:
That's a great idea!

bajtos commented:
@raymondfeng thank you for a great suggestion!

In 586bfd8, I improved the setup scripts & incorporated review feedback.

In 824e62d, I added two npm scripts (`npm run preview-docs:prepare` and `npm run preview-docs:start`) and updated documentation for developers.

@strongloop/loopback-maintainers LGTY?

bajtos commented:
Few more changes:
- 98268e3 is configuring Prettier to ignore files in site preview
- eda5e11d4 is configuring Eslint to ingore files in site preview

raymondfeng commented on package.json:
```diff
@@ -59,6 +59,8 @@
     "verify:docs": "npm run build:site -- --verify",
     "build:site": "./bin/build-docs-site.sh",
+    "docs-preview:prepare": "./docs/bin/build-preview-site.sh",
+    "docs-preview:start": "cd docs/_preview && bundle exec jekyll serve --no-w --i",
```
docs:preview?

raymondfeng commented on package.json:
```diff
@@ -59,6 +59,8 @@
     "test:ci": "node packages/build/bin/run-nyc npm run mocha --scripts-prepend-node-path",
     "verify:docs": "npm run build:site -- --verify",
     "build:site": "./bin/build-docs-site.sh",
+    "docs-preview:prepare": "./docs/bin/build-preview-site.sh",
```
docs:prepare?

bajtos commented:
Renamed the npm scripts in c7f6f26c0.
